common/src/main/java/com/google/auto/common/MoreTypes.java

```diff
@@ -205,9 +243,40 @@ private static boolean equal(TypeMirror a, TypeMirror b, Set<ComparedElements> v
     EqualVisitorParam p = new EqualVisitorParam();
     p.type = b;
     p.visiting = visiting;
+    if (INTERSECTION_TYPE != null) {
+      if (INTERSECTION_TYPE.isInstance(a)) {
```

Comment:
Hmm, that is a good point. Indeed there are other methods in this class whose entire raison d'être is to handle the theoretical case where a TypeMirror might be instanceof something without actually representing that something. I will make a later change that uses a.getKind().name() instead. That will allow us to continue to work on pre-Java-8 without requiring the introduction of yet more reflection objects.